capsules/src/process_console.rs

```diff
@@ -967,6 +996,80 @@ impl<'a, A: Alarm<'a>, C: ProcessManagementCapability> uart::ReceiveClient
+                                            None
+                                        } else {
+                                            //check if there even is a command to move up to
+                                            self.command_history
+                                                .map(|cmd_arr| match cmd_arr[i][0] {
+                                                    0 => None,
+                                                    _ => Some(i),
+                                                })
+                                                .unwrap()
+                                        }
+                                    }
+                                    //down arrow case
+                                    b'B' => {
+                                        if self.command_history_index.is_some()
+                                            && self.command_history_index.extract().unwrap() > 0
+                                        {
+                                            Some(self.command_history_index.extract().unwrap() - 1)
+                                        } else {
+                                            None
+                                        }
+                                    }
+                                    _ => None,
+                                } {
+                                    self.command_history_index.set(index);
+                                    self.command_history.map(|next_command| {
+                                        let prev_command_len = self.command_index.get();
+                                        let next_command_len = {
+                                            let mut i = 0;
+                                            loop {
+                                                if next_command[index][i] == 0 {
+                                                    break i;
+                                                }
+                                                i = i + 1;
+                                            }
+                                        };
+
+                                        //remove the last command from screen
+                                        for _ in 0..prev_command_len {
+                                            let _ = self.write_bytes(&[
+                                                '\x08' as u8,
+                                                ' ' as u8,
+                                                '\x08' as u8,
+                                            ]);
+                                        }
+                                        //and copy the next one
+                                        for i in 0..next_command_len {
+                                            let _ = self.write_byte(next_command[index][i]);
+                                            command[i] = next_command[index][i];
```

Comment:
I don't know if it is a github error, but the "[i]" parameter is displayed on the next line.

Reply:
Github's error